packages/actions-shared/src/engage/utils/EngageLogger.ts

```
@@ -25,33 +25,53 @@ export class EngageLogger extends OperationLogger {
   readonly logDetails: Record<string, unknown> = {}
 
   logInfo(msg: string, metadata?: object): void {
-    const msgs = [msg, ...(metadata ? [JSON.stringify(metadata)] : [])]
     if (!this.actionPerformer.isFeatureActive(FLAGON_NAME_LOG_INFO, () => false)) return
-    const [firstMsg, ...rest] = msgs
-    this.loggerClient?.info(
-      `TE Messaging: ${this.channelType} ${firstMsg}`,
-      ...rest,
-      JSON.stringify({ ...this.logDetails, ...metadata })
-    )
+    try {
+      const msgs = [msg, ...(metadata ? [JSON.stringify(metadata)] : [])]
+      const [firstMsg, ...rest] = msgs
+      this.loggerClient?.info(
+        `TE Messaging: ${this.channelType} ${firstMsg}`,
+        ...rest,
+        JSON.stringify({ ...this.logDetails, ...metadata })
+      )
+    } catch {
```

Could we remove these two try/catch blocks then? Wouldn't it be easier to debug if we were throwing when an error occurred rather than completely ignoring the error?

---

no, we have unreliable behavior of logging, specifically dropped logs during high load. It's still not clear why logs are dropped, infra-observability is aware of it and could not help (see rfc doc for details). This empty catch is to make execution logger-fail-proof